bgpd/bgp_evpn_vty.c

```diff
@@ -6639,9 +6639,7 @@ void bgp_config_write_evpn_info(struct vty *vty, struct bgp *bgp, afi_t afi,
 				char *vni_str = NULL;
 
 				vni_str = strchr(ecom_str, ':');
-
-				if (!vni_str)
-					continue; /* This should never happen */
+				assert(vni_str);
```

Comment:
I would really prefer that we don't crash the system on a mistake.  What's the goal here that you are trying to accomplish?  Why do we need this change?

Reply:
It's impossible case.  I think `ecom_str` **here** must absolutely have `:`.

Reply:
What are you trying to accomplish with this change?  The impossible situation is handled already and you are changing it from something benign to a full on stop of the program.  Why does FRR need this?

Reply:
I think he is saying if we ever hit this impossible situation it will technically leak. I imagine a static analyzer may pick up on that.

just put an XFREE + continue here or a goto free

Reply:
Thanks, done.